src/gui/qgslockbutton.cpp

```diff
@@ -0,0 +1,117 @@
+/***************************************************************************
+    qgslockbutton.cpp - Lock button
+     --------------------------------------
+    Date                 : July, 2017
+    Copyright            : (C) 2017 by Mathieu Pellerin
+    Email                : nirvn dot asia at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslockbutton.h"
+
+#include <QApplication>
+#include <QMouseEvent>
+#include <QPainter>
+#include <QPushButton>
+#include <QWidget>
+
+QgsLockButton::QgsLockButton( QWidget *parent )
+  : QToolButton( parent )
+  , mLocked( false )
+
+{
+  setMinimumSize( QSize( 24, 24 ) );
+  setCheckable( true );
```

Comment:
Try a 

    setAutoRaise( true );

Here too - that should get the button appearance a bit closer to GIMP's